api/api.go

```diff
@@ -0,0 +1,112 @@
+package api
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/stretchr/graceful"
+)
+
+const (
+	apiTimeout = 5 * time.Second
+)
+
+// handleHTTPRequest is a wrapper function that logs and then handles all
+// incoming calls to the API.
+func handleHTTPRequest(mux *http.ServeMux, url string, handler http.HandlerFunc) {
+	mux.HandleFunc(url, func(w http.ResponseWriter, req *http.Request) {
+		log.Printf("%s %s", req.Method, req.URL)
+		handler(w, req)
+	})
+}
+
+// initAPI determines which functions handle each API call.
+func (srv *Server) initAPI(addr string) {
+	mux := http.NewServeMux()
+
+	// Consensus API Calls
+	handleHTTPRequest(mux, "/consensus/status", srv.consensusStatusHandler)
+
+	// Server API Calls
+	handleHTTPRequest(mux, "/daemon/stop", srv.daemonStopHandler)
+	handleHTTPRequest(mux, "/daemon/update/apply", srv.daemonUpdateApplyHandler)
+	handleHTTPRequest(mux, "/daemon/update/check", srv.daemonUpdateCheckHandler)
```

Comment:
these should all be renamed to /server, and the function names should be srv.serverXXXHandler